### PR TITLE
fix archive module can not compress a single file in a tar file

### LINKED ISSUE #44968: archive module can not compress a single file in a tar file
##### SUMMARY
The archive module can not compress a single file in a tar file getting the following error:
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_lzBroV/ansible_module_archive.py", line 440, in main
    raise OSError("Invalid format")
OSError: Invalid format
```

##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
archive

##### ANSIBLE VERSION
```
ansible 2.6.3
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]

```

##### OS / ENVIRONMENT
N/A

##### STEPS TO REPRODUCE
Playbook:
```
- name: Compressing file
  hosts: all
  
  tasks:
    - name: Archive module
      archive:
        path: "/tmp/test_file.txt"
        dest: "/tmp/test.tar"
        format: tar
```

##### EXPECTED RESULTS
The "archive" module should compress files in tar files.

##### ACTUAL RESULTS
Error:
```
The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_lzBroV/ansible_module_archive.py", line 440, in main
    raise OSError("Invalid format")
OSError: Invalid format

fatal: [XX.XX.X.XX]: FAILED! => {
    "changed": false, 
    "dest": "/tmp/test.tar", 
    "gid": 0, 
    "group": "root", 
    "invocation": {
        "module_args": {
            "attributes": null, 
            "backup": null, 
            "content": null, 
            "delimiter": null, 
            "dest": "/tmp/test.tar", 
            "directory_mode": null, 
            "exclude_path": null, 
            "follow": false, 
            "force": null, 
            "format": "tar", 
            "group": null, 
            "mode": null, 
            "owner": null, 
            "path": [
                "/tmp/test_file.txt"
            ], 
            "regexp": null, 
            "remote_src": null, 
            "remove": false, 
            "selevel": null, 
            "serole": null, 
            "setype": null, 
            "seuser": null, 
            "src": null, 
            "unsafe_writes": null
        }
    }, 
    "mode": "0644", 
    "msg": "Unable to write to compressed file: Invalid format", 
    "owner": "root", 
    "path": "/tmp/fichero_prueba.txt", 
    "secontext": "unconfined_u:object_r:user_tmp_t:s0", 
    "size": 0, 
    "state": "file", 
    "uid": 900
}
```

### PR DESCRIPTION
**Pull request by giovannisciortino**
##### SUMMARY
Before this commit the archive module source code doesn't contains the code to create a tar archive of a single file. Before this PR an OSError exception is [raised](https://github.com/ansible/ansible/blob/5c9241fa7c60f3aa731368971d4e0a481bbf63d6/lib/ansible/modules/files/archive.py#L440) because the code included between the lines 424 and 438 doesn't manage the case "format == 'tar'".

I have also verified that 'tar' is a valid value for the [format](https://github.com/ansible/ansible/blob/5c9241fa7c60f3aa731368971d4e0a481bbf63d6/lib/ansible/modules/files/archive.py#L34) parameter.

The code added to manage single file tar archives is similar to the same code present in the module used to manage multiple file tar archives starting from line [319](https://github.com/ansible/ansible/blob/5c9241fa7c60f3aa731368971d4e0a481bbf63d6/lib/ansible/modules/files/archive.py#L319).

Fixes #44968

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
archive

##### ANSIBLE VERSION
```
$ ./bin/ansible --version
ansible 2.8.0.dev0 (pull-issue-44968 a793a3d109) last updated 2018/09/03 15:58:34 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/gsciorti/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/gsciorti/PycharmProjects/ansible/lib/ansible
  executable location = ./bin/ansible
  python version = 2.7.14 (default, Dec 11 2017, 14:52:53) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

##### ADDITIONAL INFORMATION

**BEFORE THIS COMMIT**
```
$ ./bin/ansible localhost -m archive -a "path=/etc/hosts dest=/tmp/hosts.tar format=tar"
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: OSError: Invalid format
localhost | FAILED! => {
    "changed": false, 
    "dest": "/tmp/hosts.tar", 
    "gid": 0, 
    "group": "root", 
    "mode": "0644", 
    "msg": "Unable to write to compressed file: Invalid format", 
    "owner": "root", 
    "path": "/etc/hosts", 
    "secontext": "system_u:object_r:net_conf_t:s0", 
    "size": 4552, 
    "state": "file", 
    "uid": 0
}
```

**AFTER THIS COMMIT**
```
$ ./bin/ansible localhost -m archive -a "path=/etc/hosts dest=/tmp/hosts.tar format=tar"
localhost | CHANGED => {
    "archived": [
        "/etc/hosts"
    ], 
    "arcroot": "/etc/", 
    "changed": true, 
    "dest": "/tmp/hosts.tar", 
    "expanded_exclude_paths": [], 
    "expanded_paths": [
        "/etc/hosts"
    ], 
    "gid": 1000, 
    "group": "gsciorti", ```

    "missing": [], 
    "mode": "0664", 
    "owner": "gsciorti", 
    "secontext": "unconfined_u:object_r:user_tmp_t:s0", 
    "size": 10240, 
    "state": "file", 
    "uid": 1000
}
```

### CODE REVIEW

**Comment by bendoh** [comment]
shipit

**Comment by robertdebock** [comment]
Tested it locally, see #44968.

TL;DR: This fix works!
